tools/amd_build/pyHIPIFY/hipify-python.py

```diff
@@ -0,0 +1,946 @@
+                    if status == AT_KERNEL_NAME:
+                        pos["kernel_name"]["start"] = i
+
+                        # Finished
+                        return [(pos["kernel_name"]), (pos["template"]), (pos["kernel_launch"])]
+
+
+    def find_kernel_bounds(string):
+        """Finds the starting and ending points for all kernel launches in the string."""
+        kernel_end = 0
+        kernel_positions = []
+
+        # Continue until we cannot find any more kernels anymore.
+        while string.find("<<<", kernel_end) != -1:
+            # Get kernel starting position (starting from the previous ending point)
+            kernel_start = string.find("<<<", kernel_end)
+
+            # Get kernel ending position (adjust end point past the >>>)
+            kernel_end = string.find(">>>", kernel_start) + 3
+
+            # Add to list of traversed kernels
+            kernel_positions.append({"start": kernel_start, "end": kernel_end, "group": string[kernel_start: kernel_end]})
+
+        return kernel_positions
+
+
+    # Grab positional ranges of all kernel launchces
+    get_kernel_positions = [k for k in find_kernel_bounds(string)]
+    output_string = string
+
+    # Replace each CUDA kernel with a HIP kernel.
+    for kernel in get_kernel_positions:
+        # Get kernel components
+        params = grab_method_and_template(kernel)
+
+        # Find paranthesis after kernel launch
+        paranthesis = string.find("(", kernel["end"])
+
+        # Extract cuda kernel
+        cuda_kernel = string[params[0]["start"]:paranthesis+1]
+
+        # Keep number of kernel launch params consistent (grid dims, group dims, stream, dynamic shared size)
+        num_klp = len(extract_arguments(0, kernel["group"].replace("<<<", "(").replace(">>>", ")")))
+
+        # Transform cuda kernel to hip kernel
+        hip_kernel = "hipLaunchKernelGGL(" + cuda_kernel[0:-1].replace(">>>", ", 0"*(4-num_klp) + ">>>").replace("<<<", ", ").replace(">>>", ", ")
+
+        # Replace cuda kernel with hip kernel
+        output_string = output_string.replace(cuda_kernel, hip_kernel)
+
+        # Update the statistics
+        stats["kernel_launches"].append(hip_kernel)
+
+    return output_string
+
+
+def find_paranthesis_end(input_string, start):
+    inside_paranthesis = False
+    parans = 0
+    pos = start
+    p_start, p_end = -1, -1
+
+    while pos < len(input_string):
+        if input_string[pos] == "(":
+            if inside_paranthesis is False:
+                inside_paranthesis = True
+                parans = 1
+                p_start = pos
+            else:
+                parans += 1
+        elif input_string[pos] == ")" and inside_paranthesis:
+            parans -= 1
+
+            if parans == 0:
+                p_end = pos
+                return p_start, p_end
+
+        pos += 1
+    return None, None
+
+
+def disable_asserts(input_string):
+    """ Disables regular assert statements
```

Comment:
Why?

Reply:
There's a HIP bug that prevents asserts from compiling correctly.

Reply:
HIP would throw compilation errors whenever we used assert calls. Though, this may or may not have been resolved in the newly released versions.

Reply:
This change is still valid/necessary.